cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -162,6 +170,11 @@ protected void startUp() throws Exception {
         )
     ).get();
 
+    // Start SCM service if the feature is enabled.
```

Comment:
nit:
We can have a list of `start()` and append `sourceControlManagementService.start()` based on feature flag

Also I don't think we really need to guard `sourceControlManagementService.start()` with feature flag as long as handlers are blocked

Reply:
There is only one such optional service presently, so IMO introducing a list seems unnecessary.
Also I think we shouldn't start a service if its not needed so that we can use the feature flag to completely avoid running code paths for SCM.
Imagine we found a memory leak in the new service and we don't have a way to disable it using configuration, then we would require a new version to be released with the fix/disabling change.

Reply:
Got it, removed the feature flag checks.